ibis/omniscidb/client.py

```diff
@@ -973,7 +1036,8 @@ def database(self, name=None):
                 database=name,
                 protocol=self.protocol,
                 session_id=self.session_id,
-                execution_type=self.execution_type,
+                ipc=self.ipc,
```

Comment:
I would mock this out to handle the various scenarios

Reply:
@jreback could you explain me more your idea about this? thanks

Reply:
the only way to actually test these things is to use a https://github.com/pytest-dev/pytest-mock, pls add a set of tests that exercise your interface. basically you have added a ton of code that is untested; it may work but can't really see.